### PR TITLE
openroad-gui: add `OpenROAD` build with `GUI` option

### PR DESCRIPTION
**Pull request by joennlae**
Thank you for this excellent repo 💯 . It saved me a ton of time 👍 

This PR adds a GUI-supported build for `OpenROAD`. This build is tested and can be installed via `conda`:

* [openroad-gui](https://anaconda.org/joennlae/openroad-gui)

```bash
conda install -c joennlae/label/ci-add-openroad-gui-2771065658 openroad-gui
```

### CODE REVIEW

**Comment by proppy** [comment]
@joennlae Thanks for your contribution!

We had discussion on https://github.com/hdl/conda-eda/issues/193#issuecomment-1084035801 on the differentiation between the package in this repository and the the one in conda-forge.

I'm curious if you'd be open to create a similar PR against https://github.com/conda-forge/staged-recipes and put me as a co-maintainer?

If we also want to keep the recipe in this repo, I think it might be better to have it as a package variant `openroad.gui` in https://github.com/hdl/conda-eda/tree/master/pnr/openroad (and share the build script/recipe) rather than a totally separate package, what do you think>

**Comment by joennlae** [comment]
@proppy I completely get your point.

1) How would you approach doing it as a variant instead of a package when they have a different `build.sh`?

2) Regarding the `conda-forge`. I, unfortunately, do not have the capacity to do that atm. I see some hurdles:
- No official .tar or .zip releases of OpenROAD
- OpenROAD has a ton of different licenses.
- I really like the daily builds as the development of OpenROAD is still very fast-paced.

**Comment by proppy** [comment]
> How would you approach doing it as a variant instead of a package when they have a different build.sh?

I think we could have the `.gui` variant as a subpackage by defining one additional output https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#subpackage-requirements with a different `build` script (or one that parameterize the existing one, if it's not diverging too much).

> No official .tar or .zip releases of OpenROAD

Good point, we'd need to convince them to tag more version; one thing we discussed before with them was to tag things similar to the verible repo: https://github.com/chipsalliance/verible/tags

> OpenROAD has a ton of different licenses.

I think that would also be a proble with conda-eda, I was under the impression that all of OpenROAD was under BSD-3 if that's not the case we should update https://github.com/hdl/conda-eda/blob/master/pnr/openroad/meta.yaml#L60

> I really like the daily builds as the development of OpenROAD is still very fast-paced.

conda-forge has some automation to bump packages everytime there is a release, see for e.g: `magic` https://github.com/conda-forge/magic-feedstock/pulls?q=is%3Apr+is%3Aclosed+author%3Aregro-cf-autotick-bot+ but I agree the frequency will be (back to your first point) very depending on how often OpenROAD tag a release.

**Comment by joennlae** [comment]
Ok. If you help me move forward on the `conda-forge` front. I am happy to support you 👍  Regarding the subpackage. We can do that if you like it more that way.

**Comment by proppy** [comment]
@joennlae sorry for the late reply.

> Ok. If you help me move forward on the conda-forge front. I am happy to support you 

Would it be OK for us to re-submit your work here as a package on https://github.com/conda-forge/staged-recipes and add us both as co-maintainers?

> +1 Regarding the subpackage. We can do that if you like it more that way.

Let me know if you have any questions, also happy to send a PR against your branch to get this started.

**Comment by proppy** [comment]
> > No official .tar or .zip releases of OpenROAD
> 
> Good point, we'd need to convince them to tag more version; one thing we discussed before with them was to tag things similar to the verible repo: https://github.com/chipsalliance/verible/tags

Started a discussion around this here:
https://github.com/The-OpenROAD-Project/OpenROAD/issues/2153

File: pnr/openroad-gui/condarc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+channels:
+  - conda-forge
```
Comment (proppy) [inline pnr/openroad-gui/condarc:2]: curious why conda-forge is needed?

File: pnr/openroad-gui/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    - libboost 1.73
+    - eigen 3.3
+    - spdlog 1.8.1
+    - readline
+    - tk
+    - zlib
+    - fmt
+    - libgomp
+    - qt
+    - mesalib
```
Comment (proppy) [inline pnr/openroad-gui/meta.yaml:50]: I think we can use https://anaconda.org/main/mesa-libgl-devel-cos6-x86_64 as a `cdt` instead, see https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#:~:text=On%20Linux%2C%20using,by%20them%20automatically.

File: pnr/openroad-gui/condarc (hunk 1 above)
Comment (joennlae) [reply pnr/openroad-gui/condarc:2]: If I remember correctly, it was for qt.

File: pnr/openroad-gui/meta.yaml (hunk 2 above)
Comment (joennlae) [reply pnr/openroad-gui/meta.yaml:50]: I agree :-)

File: pnr/openroad-gui/condarc (hunk 1 above)
Comment (proppy) [reply pnr/openroad-gui/condarc]: managed to drop the dependency on conda-forge by using cdt.

**Comment by proppy** [comment]
ended up adding the GUI to the default package, as @xobs pointed out this is what most folks expect by default.

**Comment by proppy** [comment]
build failing with GL.h
```
CMake Error at /Users/runner/work/conda-eda/conda-eda/workdir/conda-env/conda-bld/openroad_1677010546348/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake:9 (message):
    Failed to find "gl.h" in
    "/opt/MacOSX10.11.sdk/System/Library/Frameworks/OpenGL.framework/Headers;/opt/MacOSX10.11.sdk/System/Library/Frameworks/AGL.framework/Headers".
  Call Stack (most recent call first):
    /Users/runner/work/conda-eda/conda-eda/workdir/conda-env/conda-bld/openroad_1677010546348/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/cmake/Qt5Gui/Qt5GuiConfig.cmake:171 (include)
    /Users/runner/work/conda-eda/conda-eda/workdir/conda-env/conda-bld/openroad_1677010546348/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/cmake/Qt5Widgets/Qt5WidgetsConfig.cmake:88 (find_package)
    /Users/runner/work/conda-eda/conda-eda/workdir/conda-env/conda-bld/openroad_1677010546348/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/cmake/Qt5/Qt5Config.cmake:28 (find_package)
    src/gui/CMakeLists.txt:9 (find_package)
```

**Comment by proppy** [comment]
curious why an older version `5.9.7` of qt get selected rather than the latest one available in https://anaconda.org/main/qt/files.

**Comment by proppy** [comment]
@xobs skipped the ui for now on osx, does this look good to you?

**Comment by proppy** [comment]
Auditing the warning from the error logs:
```
2023-02-22T04:39:17.3893774Z [32;1m04:39:17[0m | Warning: rpath /root/conda-eda/conda-eda/workdir/conda-env/conda-bld/openroad_1677039184101/_build_env/lib is outside prefix /root/conda-eda/conda-eda/workdir/conda-env/conda-bld/openroad_1677039184101/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p (removing it)
```
seems safe to ignore since it's just about cleaning up the rpath.

```
2023-02-22T04:39:17.3925894Z [32;1m04:39:17[0m | WARNING :: Failed to get_static_lib_exports(/root/conda-eda/conda-eda/workdir/conda-env/conda-bld/openroad_1677039184101/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/libxcb-xv.a)
```
safe to ignore per @ajelinski in https://github.com/hdl/conda-eda/pull/165#issuecomment-1059282395


```
2023-02-22T04:39:17.4155044Z [32;1m04:39:17[0m | WARNING (openroad): run-exports library package pkgs/main::readline-8.2-h5eee18b_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```
should probably remove readline from meta.

**Comment by xobs** [comment]
Looks good to me! I'll test it once it's merged and see how difficult it would be to at least build on my local osx-arm64 machine.

**Comment by proppy** [comment]
found another warning:
```
2023-02-22T06:33:30.4062078Z [32;1m06:33:30[0m | WARNING (openroad,bin/openroad): Needed DSO lib/libfmt.so.8 found in ['fmt']
2023-02-22T06:33:30.4062777Z [32;1m06:33:30[0m | WARNING (openroad,bin/openroad): .. but ['fmt'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

**Comment by proppy** [comment]
the `fmt` warning is expected since `fmt` is an header-only library, see
```
  ignore_run_exports_from:
    # header-only libraries
    - fmt
```

**Comment by proppy** [comment]
merging since build are passing on both plateforms.

**Comment by b224hisl** [comment]
Excuse me， I still can't use the gui even I have installed this conda packages
image

**Pull request merged**